File: app-miyou/src/main/java/com/miyou/app/domain/credit/model/ConversationDeduction.java
package com.miyou.app.domain.credit.model;

import com.miyou.app.domain.dialogue.model.ConversationSessionId;

public record ConversationDeduction(
	ConversationSessionId sessionId) implements CreditSource {

	@Override
	public CreditSourceType sourceType() {
		return CreditSourceType.CONVERSATION_DEDUCTION;
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/credit/model/ConversationDeduction.kt
package com.miyou.app.domain.credit.model

import com.miyou.app.domain.dialogue.model.ConversationSessionId

data class ConversationDeduction(
    val sessionId: ConversationSessionId,
) : CreditSource {
    override fun sourceType(): CreditSourceType = CreditSourceType.CONVERSATION_DEDUCTION

    fun sessionId(): ConversationSessionId = sessionId
}
